usr.sbin/devinfo/devinfo.c

```diff
@@ -57,6 +57,18 @@ struct indent_arg
 	void	*arg;
 };
 
+
+static void
+print_indent(int n)
+{
+	char *s;
+
+	if (n < 1)
+		return;
+	asprintf(&s, "%*s", n, " ");
```

Comment:
`asprintf()` causes allocation, but it doesn't seem like there's any need for this?
equivalent to
```
	while (n--)
		putchar(' ');
```

Reply:
It's done this way to minimize diff when libxo is added later. With libxo we want to emit the whole padding in one call. Otherwise the html get spammed with <div class="padding"> elements.

Reply:
still, you need to free(s) here...

Reply:
Fixed. Along with a few other cases. Sorry, I'm not used to C.